components/crypto/crypto-softhsm-impl/src/integrationTest/kotlin/repository/WrappingRepositoryTest.kt

```diff
@@ -78,4 +79,42 @@ class WrappingRepositoryTest : CryptoRepositoryTest() {
         assertThat(loadedKey).isNull()
     }
 
-}
\ No newline at end of file
+    @ParameterizedTest
+    @MethodSource("emfs")
+    fun `findKeyWithId returns the UUID and the key`(emf: EntityManagerFactory) {
+        val keyAlias = "find-key-${UUID.randomUUID()}"
+        val repo = WrappingRepositoryImpl(emf, "test")
+        repo.saveKey(keyAlias, wrappingKeyInfo)
+        val loadedKeyAndId = repo.findKeyAndId(keyAlias)
+
+        if (loadedKeyAndId != null) {
+            assertInstanceOf(UUID::class.java, loadedKeyAndId.first)
+            assertThat(loadedKeyAndId.second).isEqualTo(wrappingKeyInfo)
+        }
```

Comment:
Do you not need an additional assert to fail in case `loadedKeyAndId` is null? E.g.

```suggestion
        assertThat(loadedKeyAndId).isNotNull()
        assertInstanceOf(UUID::class.java, loadedKeyAndId??.first)
        assertThat(loadedKeyAndId.second).isEqualTo(wrappingKeyInfo)
```

Reply:
Yeah, it makes sense to have the not null check!